components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+			local link = input:gsub(' ', '_')
+			local data = {link = link}
+
+			-- Automatically retrieve information from the Tournament
+			local tournamentData = prizePool:_getTournamentInfo(link)
+			if tournamentData then
+				data.title = tournamentData.tickername
+				data.icon = tournamentData.icon
+				data.iconDark = tournamentData.icondark
+			end
+
+			-- Manual override
+			local prefix = 'qualifies' .. index
+			data.title = context[prefix .. 'name'] or data.title
+
+			return data
+		end,
+		row = 'qualified',
+		rowParse = function (prizePool, input, context, index)
+			return Logic.readBool(input)
+		end
+	},
+	[prizeTypes.POINTS] = {
+		header = 'points',
+		headerParse = function (prizePool, input, context, index)
+			local pointsData = mw.loadData('Module:Points/data')
+			return pointsData[input] or {title = 'Points'}
+		end,
+		row = 'points',
+		rowParse = function (prizePool, input, context, index)
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.FREETEXT] = {
+		header = 'freetext',
+		headerParse = function (prizePool, input, context, index)
+			return {title = input}
+		end,
+		row = 'freetext',
+		rowParse = function (prizePool, input, context, index)
+			return input
+		end
+	},
+}
+
+local specialData = {
+	[specialDataTypes.GROUPSCORE] = {
+		field = 'wdl',
+		parse = function (prizePool, input, context)
+			return input
+		end
+	},
+	[specialDataTypes.LASTVS] = {
+		field = 'lastvs',
+		parse = function (prizePool, input, context)
+			return prizePool:_parseOpponentArgs(input, context.date)
+		end
+	},
+	[specialDataTypes.LASTVSSCORE] = {
+		field = 'lastvsscore',
+		parse = function (prizePool, input, context)
+			local scores = Table.mapValues(mw.text.split(input, '-'), tonumber)
+			return {score = scores[1], vsscore = scores[2]}
+		end
+	},
+}
+
+function PrizePool:init(args)
+	self.args = args
+	self.pagename = mw.title.getCurrentTitle().text
+	self.args.type = Json.parseIfString(self.args.type)
+	self.date = args.date or Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
+	if args.opponentLibrary then
+		Opponent = require('Module:'.. args.opponentLibrary)
+	end
+
+	if not self.args.type then
+		return error('Please provide a type!')
+	elseif type(self.args.type) ~= 'table' or not self.args.type.type then
+		return error('Could not parse type!')
+	elseif not Opponent.isType(self.args.type.type) then
+		return error('Not a valid type!')
+	end
+
+	self.opponentType = self.args.type.type
+
+	return self
+end
+
+function PrizePool:readInput(args)
+	self.options = self:_readStandardOptions(args)
+	self.prizes = self:_readPrizes(args)
+	self.placements = self:_readPlacements(args)
+
+	if self:_hasUsdPrizePool() then
+		self:setOption('showUSD', true)
+		self:addPrize('USD', 1)
+	end
+
+	return self
+end
+
+function PrizePool:create()
+	-- TODO
+	mw.logObject(self)
+	return ''
+end
+
+function PrizePool:_readStandardOptions(args)
```

Comment:
What are "standard" options? This seems like a sort of config, would that not be a better name? Something along the lines of: `getDefaultConfig`?

Reply:
Renamed and reworked.